variables.tf

```diff
@@ -9,3 +13,257 @@ variable "tags" {
   description = "User-Defined tags"
   type        = map(string)
 }
+
+########################################
+# Bitbucket Vars
+########################################
+
+variable "admin_email" {
+  description = "email address for administrator"
+  type        = string
+}
+
+variable "admin_password" {
+  description = "initial password to use for administrator"
+  type        = string
+}
+
+variable "data_volume_size" {
+  description = "Size in GB of the data volume"
+  type        = number
+}
+
+variable "dns_prefix" {
+  default     = null
+  description = "the hostname that will be used for bitbucket. This will be combined with the domain in `zone_id` or the value of `domain_name` to form the base url."
+  type        = string
+}
+
+variable "domain_name" {
+  default     = null
+  description = "domain name, which is only used if `zone_id` is not specified to compute the base url"
+  type        = string
+}
+
+variable "license_key" {
+  default     = ""
+  description = "Bitbucket license key (optional, must be a single line)"
+  type        = string
+}
+
+variable "site_name" {
+  default     = "Bitbucket"
+  description = "Bitbucket site name"
+  type        = string
+}
+
+variable "volume_key" {
+  default     = "bitbucket-volume"
+  description = "This value is set to a key on the EBS volume and must be present for the bitbucket instance to be permitted to attach it."
+  type        = string
+}
+
+########################################
+# ASG Vars
+########################################
+
+variable "ami_id" {
+  description = "AMI to build on (must have `ansible-role-atlassian-bitbucket` module installed)"
+  type        = string
+}
+
+variable "asg_additional_iam_policies" {
+  default     = []
+  description = "Additional IAM policies to attach to the  ASG instance profile"
+  type        = list(string)
+}
+
+variable "asg_additional_security_groups" {
+  default     = []
+  description = "Additional security group IDs to attach to ASG instances"
+  type        = list(string)
+}
+
+variable "asg_additional_user_data" {
+  default     = ""
+  description = "Additional User Data to attach to the launch template"
+  type        = string
```

Comment:
We could make this a map if we want to allow multiple scripts to be passed.

Reply:
Potential future improvement